CMAKE Configuration Test Failures for clang 3.5 on debian

Configuration test failures shown for c++11 feature tests, shown here, https://gist.github.com/vamatya/e73c9454944b0f97bc33, however the tests seem actually not performed : https://gist.github.com/vamatya/ed6a75bef5d96ee0ec7b. 

hpx_version: commit 41e1e53658e0b6c463a38e020e7e7f149fe7f00f ; May 25
platform: debian, 
compiler: clang 3.5

Am 30.05.2015 1:32 nachm. schrieb "Vinay Amatya" <notifications@github.com>:
>
> printenv: https://gist.github.com/vamatya/c1d49ae88ce1a48e2bc4
> cmake_out after applying patch :
https://gist.github.com/vamatya/597291880985c7212b35

There you go. The boost libraries can not be linked (undefined reference to
some std library). How did you build those? Did you link the boost libs
with libc++? Could you rebuild them with clang in C++14 mode?

>
> Indiviually compiling all c++11_test files with clang/clang++ went
through except, cxx11_std_is_final.cpp

Sure, because when you build them just like that, you probably don't link
against the boost libraries.
This is a bug in our cmake scripts (that is, the compiler feature tests
shouldn't link against boost). However, you would have run into the same
error when linking libhpx.

>
> —
> Reply to this email directly or view it on GitHub.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -551,113 +551,6 @@ include_directories("${PROJECT_SOURCE_DIR}" "${CMAKE_BINARY_DIR}")
 link_directories(${CMAKE_BINARY_DIR}/lib)
 
 ################################################################################
-
-##############################################################################
-# Find Our dependencies:
-#   These are all dependencies needed to build the core library. Dependencies
-#   that are only needed by plugins, examples or tests should be found
-#   seperately in the appropriate subdirectory.
-# When a package is found, the necessary include paths are set, and the libraries
-# are added to the HPX_LIBRARIES variables
-
-# Setting up our required Boost libraries. This will find all required boost
-# libraries and add possibly needed boost headers shipped with the hpx source
-include(HPX_SetupBoost)
-
-# Find all allocators which are currently supported.
-include(HPX_SetupAllocator)
-
-# convey selected allocator type to the build configuration
-hpx_add_config_define(HPX_HAVE_MALLOC "\"${HPX_WITH_MALLOC}\"")
-
-if(HPX_WITH_HWLOC)
-  find_package(Hwloc)
-  if(NOT HWLOC_FOUND)
-    hpx_error("Hwloc could not be found and HPX_WITH_HWLOC=ON, please specify HWLOC_ROOT to point to the correct location or set HPX_WITH_HWLOC to OFF")
-  endif()
-  hpx_libraries(${HWLOC_LIBRARIES})
-  include_directories(${HWLOC_INCLUDE_DIR})
-  hpx_add_config_define(HPX_HAVE_HWLOC)
-endif()
-
-################################################################################
-# Enable integration with Intel Amplifier and Inspector tools
-################################################################################
-if(WITH_AMPLIFIER)
-  find_package(Amplifier)
-  if(NOT AMPLIFIER_FOUND)
-    hpx_error("Intel Amplifier could not be found and WITH_AMPLIFIER=On, please specify AMPLIFIER_ROOT to point to the root of your Amplifier installation")
-  endif()
-  hpx_libraries(${AMPLIFIER_LIBRARIES})
-  include_directories(${AMPLIFIER_INCLUDE_DIR})
-  hpx_add_config_define(HPX_HAVE_ITTNOTIFY)
-endif()
-
-if(HPX_WITH_TAU)
-  find_package(TAU)
-  if(NOT TAU_FOUND)
-    hpx_error("TAU could not be found and HPX_WITH_TAU=On, please specify TAU_ROOT to point to the root of your TAU installation")
-  endif()
-  hpx_libraries(${TAU_LIBRARIES})
-  include_directories(${TAU_INCLUDE_DIR})
-endif()
-
-################################################################################
-# Enable integration with Apex event counters
-################################################################################
-if(HPX_WITH_APEX)
-  find_package(APEX)
-  if(NOT APEX_FOUND)
-    hpx_error("Apex could not be found and HPX_WITH_APEX=On, please specify APEX_ROOT to point to the root of your Apex installation")
-  endif()
-  hpx_libraries(${APEX_LIBRARIES})
-  include_directories(${APEX_INCLUDE_DIR})
-  if(AMPLIFIER_FOUND)
-    hpx_error("AMPLIFIER_FOUND has been set. Please disable the use of the Intel Amplifier (WITH_AMPLIFIER=Off) in order to use Apex")
-  endif()
-  hpx_add_config_define(HPX_HAVE_ITTNOTIFY)
-endif()
-
-if(HPX_WITH_GOOGLE_PERFTOOLS)
-  find_package(GooglePerftools)
-  if(NOT GOOGLE_PERFTOOLS_FOUND)
-    hpx_error("Google Perftools could not be found and HPX_WITH_GOOGLE_PERFTOOLS=On, please specify GOOGLE_PERFTOOLS to point to the root of your Google Perftools installation")
-  endif()
-  hpx_libraries(${GOOGLE_PERFTOOLS_LIBRARIES})
-  include_directories(${GOOGLE_PERFTOOLS_INCLUDE_DIR})
-endif()
-
-if(HPX_WITH_VALGRIND)
-  find_package(Valgrind)
-  if(NOT VALGRIND_FOUND)
-    hpx_error("Valgrind could not be found and HPX_WITH_VALGRIND=On, please specify VALGRIND_ROOT to point to the root of your Valgrind installation")
-  endif()
-  include_directories(${VALGRIND_INCLUDE_DIR})
-  hpx_add_config_define(HPX_HAVE_VALGRIND)
-endif()
-
-################################################################################
-# Security, libsodium: P(ortable|ackageable) NaCl
-################################################################################
-if(HPX_WITH_SECURITY)
-  find_package(Sodium)
-  if(NOT SODIUM_FOUND)
-    hpx_error("Sodium could not be found and WITH_SODIUM=On, please specify Sodium_ROOT to point to the root of your Sodium installation")
-  endif()
-  hpx_add_config_define(HPX_HAVE_SODIUM)
-
-  include_directories(${SODIUM_INCLUDE_DIR})
-
-  if(MSVC)
-    hpx_add_config_define(SODIUM_STATIC)
-  endif()
-  hpx_libraries(${SODIUM_LIBRARIES})
-
-  hpx_add_config_define(HPX_HAVE_SECURITY)
-endif()
-################################################################################
-
-##############################################################################
 # Check for compiler compatibility
 #
 
@@ -1110,7 +1003,112 @@ else()
   endif()
 endif()
 
-##############################################################################
+################################################################################
+
+################################################################################
+# Find Our dependencies:
+#   These are all dependencies needed to build the core library. Dependencies
+#   that are only needed by plugins, examples or tests should be found
+#   seperately in the appropriate subdirectory.
+# When a package is found, the necessary include paths are set, and the libraries
+# are added to the HPX_LIBRARIES variables
+
+# Setting up our required Boost libraries. This will find all required boost
+# libraries and add possibly needed boost headers shipped with the hpx source
+include(HPX_SetupBoost)
+
+# Find all allocators which are currently supported.
+include(HPX_SetupAllocator)
+
+# convey selected allocator type to the build configuration
+hpx_add_config_define(HPX_HAVE_MALLOC "\"${HPX_WITH_MALLOC}\"")
+
+if(HPX_WITH_HWLOC)
+  find_package(Hwloc)
+  if(NOT HWLOC_FOUND)
+    hpx_error("Hwloc could not be found and HPX_WITH_HWLOC=ON, please specify HWLOC_ROOT to point to the correct location or set HPX_WITH_HWLOC to OFF")
+  endif()
+  hpx_libraries(${HWLOC_LIBRARIES})
+  include_directories(${HWLOC_INCLUDE_DIR})
+  hpx_add_config_define(HPX_HAVE_HWLOC)
+endif()
+
+################################################################################
+# Enable integration with Intel Amplifier and Inspector tools
+################################################################################
+if(WITH_AMPLIFIER)
+  find_package(Amplifier)
+  if(NOT AMPLIFIER_FOUND)
+    hpx_error("Intel Amplifier could not be found and WITH_AMPLIFIER=On, please specify AMPLIFIER_ROOT to point to the root of your Amplifier installation")
+  endif()
+  hpx_libraries(${AMPLIFIER_LIBRARIES})
+  include_directories(${AMPLIFIER_INCLUDE_DIR})
+  hpx_add_config_define(HPX_HAVE_ITTNOTIFY)
+endif()
+
+if(HPX_WITH_TAU)
+  find_package(TAU)
+  if(NOT TAU_FOUND)
+    hpx_error("TAU could not be found and HPX_WITH_TAU=On, please specify TAU_ROOT to point to the root of your TAU installation")
+  endif()
+  hpx_libraries(${TAU_LIBRARIES})
+  include_directories(${TAU_INCLUDE_DIR})
+endif()
+
+################################################################################
+# Enable integration with Apex event counters
+################################################################################
+if(HPX_WITH_APEX)
+  find_package(APEX)
+  if(NOT APEX_FOUND)
+    hpx_error("Apex could not be found and HPX_WITH_APEX=On, please specify APEX_ROOT to point to the root of your Apex installation")
+  endif()
+  hpx_libraries(${APEX_LIBRARIES})
+  include_directories(${APEX_INCLUDE_DIR})
+  if(AMPLIFIER_FOUND)
+    hpx_error("AMPLIFIER_FOUND has been set. Please disable the use of the Intel Amplifier (WITH_AMPLIFIER=Off) in order to use Apex")
+  endif()
+  hpx_add_config_define(HPX_HAVE_ITTNOTIFY)
+endif()
+
+if(HPX_WITH_GOOGLE_PERFTOOLS)
+  find_package(GooglePerftools)
+  if(NOT GOOGLE_PERFTOOLS_FOUND)
+    hpx_error("Google Perftools could not be found and HPX_WITH_GOOGLE_PERFTOOLS=On, please specify GOOGLE_PERFTOOLS to point to the root of your Google Perftools installation")
+  endif()
+  hpx_libraries(${GOOGLE_PERFTOOLS_LIBRARIES})
+  include_directories(${GOOGLE_PERFTOOLS_INCLUDE_DIR})
+endif()
+
+if(HPX_WITH_VALGRIND)
+  find_package(Valgrind)
+  if(NOT VALGRIND_FOUND)
+    hpx_error("Valgrind could not be found and HPX_WITH_VALGRIND=On, please specify VALGRIND_ROOT to point to the root of your Valgrind installation")
+  endif()
+  include_directories(${VALGRIND_INCLUDE_DIR})
+  hpx_add_config_define(HPX_HAVE_VALGRIND)
+endif()
+
+################################################################################
+# Security, libsodium: P(ortable|ackageable) NaCl
+################################################################################
+if(HPX_WITH_SECURITY)
+  find_package(Sodium)
+  if(NOT SODIUM_FOUND)
+    hpx_error("Sodium could not be found and WITH_SODIUM=On, please specify Sodium_ROOT to point to the root of your Sodium installation")
+  endif()
+  hpx_add_config_define(HPX_HAVE_SODIUM)
+
+  include_directories(${SODIUM_INCLUDE_DIR})
+
+  if(MSVC)
+    hpx_add_config_define(SODIUM_STATIC)
+  endif()
+  hpx_libraries(${SODIUM_LIBRARIES})
+
+  hpx_add_config_define(HPX_HAVE_SECURITY)
+endif()
+################################################################################
 
 ################################################################################
 # Check Build Options based on the found dependencies. We also check for errors

